### PR TITLE
Propagate template from EntityRepository

### PR DESCRIPTION
**Pull request by VincentLanglet**
Hi @greg0ire, seems like there is no psalm on this project
I could try to add it during some of my free time.

This PR avoid to a `TooManyTemplateParams` error from psalm and allow to benefit from the `ObjectRepository<T>` interface.

### CODE REVIEW

**Review by ostrolucky — APPROVED** [review]

**Review by dmaicher — APPROVED** [review]

File: Repository/ServiceEntityRepository.php [hunk 1]
```diff
@@ -19,6 +19,9 @@
  *         parent::__construct($registry, YourEntity::class);
  *     }
  * }
+ *
+ * @template T
+ * @template-extends EntityRepository<T>
```
Comment (stof) [inline Repository/ServiceEntityRepository.php:24]: you might want to add `@psalm-param class-string<T> $entityClass` on the constructor (check the exact name of that special type. I may have made a mistake in it as I haven't checked the doc again about it when writing this comment).

Comment (VincentLanglet) [reply Repository/ServiceEntityRepository.php:24]: Nice idea, I added it. I checked and I think that's the right name of the type.

**Review by greg0ire — APPROVED** [review]
> I could try to add it during some of my free time.

That would be much appreciated! There is a template for that [here](https://github.com/doctrine/.github/blob/2c60b2f943cf08df21be3dc733f1b92db8d24beb/workflow-templates/static-analysis.yml#L40) in case you're not already aware of that.

UPD: you must be aware already since you created https://github.com/doctrine/common/pull/920

**Comment by greg0ire** [comment]
Thanks @VincentLanglet !

**Comment by VincentLanglet** [comment]
Hi @greg0ire, what is needed for the next release ? :)

**Comment by ostrolucky** [comment]
I want to squeeze into 2.3.0 https://github.com/doctrine/DoctrineBundle/issues/1300 and ideally https://github.com/doctrine/DoctrineBundle/pull/1297 too. But there was no work on #1300 done yet. Feel free to help implement it if you would like 2.3.0 out of the door faster.

**Pull request merged**